### PR TITLE
Add Wyrmsun

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://discord.com/channels/206074078338088960/768903329429454899/975069589761306644
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* https://github.com/Andrettin/Wyrmsun/pull/213

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Mailaender** [comment]
bot, build io.github.Andrettin.Wyrmsun

**Comment by Mailaender** [comment]
bot, build io.github.Andrettin.Wyrmsun

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: what's the reason?

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: io.github.Andrettin.Wyrmsun.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+modules:
+  - shared-modules/lua5.1/lua-5.1.5.json
+  - shared-modules/physfs/physfs.json
+
+  - name: sqlite3
+    config-opts:
+      - --enable-threadsafe
+      - --enable-threads-override-locks
+    build-options:
+      - cflags: -DSQLITE_ENABLE_UNLOCK_NOTIFY=1
```
Comment (hfiguiere) [inline io.github.Andrettin.Wyrmsun.yml:26]:
```suggestion
    build-options:
        cflags: -DSQLITE_ENABLE_UNLOCK_NOTIFY=1
```

flatpak builder actually print a warning. I know it's yaml, it's confusing.

File: io.github.Andrettin.Wyrmsun.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        url: https://github.com/LuaDist/toluapp.git
+        tag: 1.0.93
+        commit: 9feb31ab8b97eda9561846508ee9e3a2caff8da7
+    cleanup:
+      - '/bin/toluapp'
+
+  - name: wyrmgus
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_INSTALL_PREFIX=/app
```
Comment (hfiguiere) [inline io.github.Andrettin.Wyrmsun.yml:59]:
```suggestion
```
this is automatically done.

File: flathub.json (hunk 1 above)
Comment (Mailaender) [reply flathub.json:2]: It fails to build https://flathub.org/builds/#/builders/51/builds/7093 and the builder gets stuck until timeout.

File: io.github.Andrettin.Wyrmsun.yml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+      - type: git
+        url: https://github.com/LuaDist/toluapp.git
+        tag: 1.0.93
+        commit: 9feb31ab8b97eda9561846508ee9e3a2caff8da7
+    cleanup:
+      - '/bin/toluapp'
+
+  - name: wyrmgus
+    buildsystem: cmake-ninja
+    config-opts:
```
Comment (hfiguiere) [inline io.github.Andrettin.Wyrmsun.yml:58]:
```suggestion
    build-options:
        arch:
            aarch64:
                config-opts:
                    -DUSE_GLES=ON
    config-opts:
```

Maybe with this it build on aarch64?

File: flathub.json (hunk 1 above)
Comment (hfiguiere) [reply flathub.json:2]: but
```
-- Found OpenGL: /usr/lib/aarch64-linux-gnu/libOpenGL.so   
-- Could not find OpenGL ES 1.1 libraries
```

File: io.github.Andrettin.Wyrmsun.yml (hunk 4 above)
Comment (Mailaender) [reply io.github.Andrettin.Wyrmsun.yml:63]: Let's give it a try.

**Comment by hfiguiere** [comment]
I guess, if it doesn't work...

File: io.github.Andrettin.Wyrmsun.yml [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+finish-args:
+  - --device=dri
+  - --share=ipc
+  - --share=network
+  - --socket=x11
+  - --socket=pulseaudio
+  - --persist=.stratagus
+
+modules:
+  - shared-modules/lua5.1/lua-5.1.5.json
```
Comment (barthalion) [inline io.github.Andrettin.Wyrmsun.yml:18]: Would it work with LuaJIT?

Comment (Mailaender) [reply io.github.Andrettin.Wyrmsun.yml:18]: There is no indication in cmake or https://github.com/Andrettin/Wyrmgus/blob/master/doc/install.html

**Comment by Mailaender** [comment]
bot, build io.github.Andrettin.Wyrmsun

**Pull request closed without merging**